src/utils/cosmosdb_model_versioned.ts

```diff
@@ -166,6 +172,66 @@ export abstract class CosmosdbModelVersioned<
     );
   }
 
+  /**
+   * Add the given ttl at all the version of the document identified by
+   * the searchKey
+   */
+
+  public updateTTLForAllVersion(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>,
+    ttl: NonNegativeNumber
+  ): TE.TaskEither<CosmosErrors, ReadonlyArray<TR>> {
+    return pipe(
+      this.findAllVersionsByPartitionKey(searchKey),
+      TE.map(RA.rights),
+      TE.chain(documentList =>
+        RA.sequence(TE.ApplicativeSeq)(
+          documentList.map(document =>
+            super.updateTTL(
+              ([document.id, searchKey[0]] as unknown) as DocumentSearchKey<
+                TR,
+                CosmosDocumentIdKey,
+                PartitionKey
+              >,
+              ttl
+            )
+          )
+        )
+      ),
```

Comment:
Maybe you can limit nesting using something like:

```suggestion
      TE.map(RA.rights),
      TE.map(
        RA.map(document =>
          super.updateTTL(
            ([document.id, searchKey[0]] as unknown) as DocumentSearchKey<
              TR,
              CosmosDocumentIdKey,
              PartitionKey
            >,
            ttl
          )
        )
      ),
      TE.chain(RA.sequence(TE.ApplicativeSeq)),
```

(I couldn't find a test for this method so I'm unsure : -)

Reply:
As mentioned in the PR description is not that easy to test this method because there is no support for `patch` by@zeit/cosmosdb-server